packages/pwa-kit-react-sdk/src/ssr/universal/components/with-react-query/index.js

```diff
@@ -51,18 +52,35 @@ export const withReactQuery = (Wrapped, options = {}) => {
             const queryClient = (res.locals.__queryClient =
                 res.locals.__queryClient || new QueryClient(queryClientConfig))
 
+            res.__performanceTimer.mark(PERFORMANCE_MARKS.reactQueryPrerender, 'start')
             // Use `ssrPrepass` to collect all uses of `useQuery`.
             await ssrPrepass(appJSX)
-
+            res.__performanceTimer.mark(PERFORMANCE_MARKS.reactQueryPrerender, 'end')
             const queryCache = queryClient.getQueryCache()
             const queries = queryCache.getAll().filter((q) => q.options.enabled !== false)
             await Promise.all(
-                queries.map((q) =>
-                    // If there's an error in this fetch, react-query will log the error
-                    q.fetch().catch(() => {
-                        // On our end, simply catch any error and move on to the next query
-                    })
-                )
+                queries.map((q, i) => {
+                    res.__performanceTimer.mark(
+                        `${PERFORMANCE_MARKS.reactQueryUseQuery}::${i}`,
```

Comment:
We are still using the index as the key? I know the cache key is large, I wanter if there are some kind of out of the box solutions we can use to we can actually trace which query the performance data is for, the index alone probably isn't ideal.

Reply:
Yes, for `getProps`, the page name is in the key. 

For `useQuery`, i'm still using an index, the query cache will be output in the console logs, but not in the http header.

Currently, the cache key is the only useful information i can use from the query, are you considering that we inject some sort of component name metadata during ssr prepass? Any ideas how that might work?

Reply:
I'm not really suggesting any solution in that response :P. Mainly stating that we are really missing the mark (no pun intended) on how we might see the customers using this performance information. If they can't identify which query is causing slowness, then whats the point. 

I was looking through the react query code and the prop called `queryMeta` .. I don't think we currently use that in `commerce-sdk-react` .. but I wonder if we are to set that value in the lib and it could also be overridden in the app using the query options, we could use it if it exists, if no fall back to the index.

Thoughts?

Reply:
added query metas to commerce-sdk-react queries!

For custom queries, default to index.